### ISSUE TITLE
Steam support broken in recent Steam versions

### ISSUE THREAD

**Issue by Zottelchen** [TanninOne/keypirinha-allmygames#13]
When trying to open a game (e.g. Elder Scrolls Online), the console prints:
```
** ERROR: Error from AllMyGames.AllMyGames.on_execute: <class 'KeyError'>: ('Steam',)
Traceback (most recent call last):
  File "C:\Users\Zottelchen\scoop\apps\keypirinha\current\portable\Profile\InstalledPackages\AllMyGames.keypirinha-package\allmygames.py", line 109, in on_execute
```
Maybe something changed with the .vdf file?

**Comment by httpsWina** [comment]
i have been struggling with the same thing. steam games simply dont show up, error is "WARNING: AllMyGames.AllMyGames: failed to initialize repo Steam Invalid magic, got b'(DV\x07'"

**Comment by TanninOne** [comment]
I can't reproduce either,

@httpsWina "Invalid magic, got b'(DV\x07'" sounds like a corrupted file somewhere, is that the entire error message?

@Zottelchen  I suspect you should have gotten another error message or warning earlier, could you have a look?

**Comment by httpsWina** [comment]
yes, this is what the logs look like after refreshing the catalog:

```
10:55:03.002 WARNING: AllMyGames.AllMyGames: failed to initialize repo Steam Invalid magic, got b'(DV\x07'
10:55:03.431 WARNING: AllMyGames.AllMyGames: [GOG] not found C:\GOG Games\SUPERHOT\SUPERHOT.exe 
10:55:03.431 WARNING: AllMyGames.AllMyGames: [GOG] not found C:\GOG Games\Project Hospital\ProjectHospital.exe
10:55:03.431 WARNING: AllMyGames.AllMyGames: [GOG] Failed to read game info [WinError 2] El sistema no puede encontrar el archivo especificado
10:55:03.438 ** ERROR: AllMyGames.AllMyGames: [UPlay] Failed to read game 274 [WinError 2] El sistema no puede encontrar el archivo especificado
10:55:03.439 AllMyGames.AllMyGames: Games found ['Epic Games Store: 1', 'Windows Store: 27', 'GOG: 0', 'Origin: 1', 'UPlay: 2']

```

the gog errors are just games i uninstalled but didnt delete the folder, and uplay idk, but i dont use it so idc.

**Comment by Zottelchen** [comment]
This looks similar to mine. I am on Steam beta branch, maybe that has something to do with it.
```
15:17:17.367 WARNING: AllMyGames.AllMyGames: failed to initialize repo Steam Invalid magic, got b'(DV\x07'
15:17:25.797 WARNING: AllMyGames.AllMyGames: [GOG] not found E:\GOG Galaxy\Games\Cyberpunk 2077\bin\x64\Cyberpunk2077.exe
15:17:25.799 WARNING: AllMyGames.AllMyGames: failed to initialize repo Origin [WinError 3] Das System kann den angegebenen Pfad nicht finden: 'C:\\ProgramData\\Origin\\LocalContent'
15:17:25.805 AllMyGames.AllMyGames: Games found ['Epic Games Store: 4', 'GOG: 0', 'UPlay: 1']
15:17:42.578 WindowsApps.WindowsApps: Cataloged 73 items in 69.9 seconds
15:17:47.429 WindowsApps.ModernControlPanel: Cataloged 91 items in 74.7 seconds
16:33:25.095 ** ERROR: Error from AllMyGames.AllMyGames.on_execute: <class 'KeyError'>: ('Steam',)
Traceback (most recent call last):
  File "C:\Users\Zottelchen\scoop\apps\keypirinha\current\portable\Profile\InstalledPackages\AllMyGames.keypirinha-package\allmygames.py", line 109, in on_execute
```

**Comment by httpsWina** [comment]
Good call, switching to the stable steam branch fixed the issue. it would be cool to see what exactly about the beta branch is causing this issue tho.

**Comment by TanninOne** [comment]
If I'm not mistaken, "b'(DV\x07'" looks like the header of binary vdf files so maybe that format changed. If so, the vdf library will require an update, unfortunately it hasn't seen an update since 2021

**Comment by Zottelchen** [comment]
Maybe this commit helps: https://github.com/SteamDatabase/SteamAppInfo/commit/b1715faba87fbbc8cb56e8fe30c92ae6ed499aab ?

**Comment by TanninOne** [comment]
Should be fixed in 1.7.0, thanks for the report and the investigation @Zottelchen
